src/coreclr/src/tools/Common/TypeSystem/Common/TargetDetails.cs

```diff
@@ -42,6 +42,10 @@ public enum TargetAbi
         /// </summary>
         CoreRT,
         /// <summary>
+        /// model for armel execution model
+        /// </summary>
+        CoreRTArmel,
```

Comment:
Did you consider making this a `TargetArchitecture` instead? A separate TargetArchitecture is [how we had it](https://github.com/dotnet/corert/commit/75ccab2fb02991cd0734ad1844df060d2aa1b010#diff-b626f7f687b5636e6788091ea3d8ff43323081a1f1b0c8897322483d2247bdd3) until it got deleted because there was no use for it at that point.

We could put a helper on `TargetDetails` to help normalize those two into one for places that don't care about the distinction.

My thinking is that while `hard`/`softfp` distinction is only about the ABI, if we ever add support for `soft` - that's not just ABI anymore. I would just treat it as a very similar, but still a bit different CPU.

Reply:
(I don't have strong opinion either way, it's just we still keep `TargetAbi.Jit` and if we ever resurrect that and have to add JitArmel for parity, it feels a bit confusing.)

Reply:
This is all very confusing, and I agree its this isn't a very stable model within the compiler. The problem is that we tend to check for architecture with a simple equality check, and there are quite a few checks (and it would be really easy to get this wrong as we don't actively test this.).

With the abi, as this change is actually specific to the abi, and we don't have many abi checks in the product, this seemed safer and less risky going forward.